File: core/src/main/java/cz/dynawest/jtexy/modules/BlockEvent.java

package cz.dynawest.jtexy.modules;

import cz.dynawest.jtexy.parsers.AroundEvent;
import cz.dynawest.jtexy.parsers.TexyParser;

/**
 *
 * @author Ondrej Zizka
 */
public class BlockEvent extends AroundEvent {

	private final String blockType;
	public String getBlockType() {		return blockType;	}
	
	private final String param;
	public String getParam() {		return param;	}



	

	public BlockEvent( TexyParser parser, String text, TexyModifier modifier, String blockType, String param ){
		super(parser, text, modifier);
		this.blockType = blockType;
		this.param = param;
	}

	public BlockEvent( BlockEvent orig, String blockType, String param ) {
		super(orig.parser, orig.text, orig.modifier);
		this.blockType = blockType;
		this.param = param;
	}




}

File: core/src/main/java/cz/dynawest/jtexy/modules/BlockEvent.kt
package cz.dynawest.jtexy.modules

import cz.dynawest.jtexy.events.AroundEvent
import cz.dynawest.jtexy.events.AroundEventListener
import cz.dynawest.jtexy.parsers.TexyParser

/**
 *
 * @author Ondrej Zizka
 */
class BlockEvent : AroundEvent {
    val blockType: String
    val param: String?

    constructor(parser: TexyParser, text: String, modifier: TexyModifier, blockType: String, param: String?) : super(
        parser,
        text,
        modifier
    ) {
        this.blockType = blockType
        this.param = param
        this.text = param
    }

    constructor(orig: BlockEvent, blockType: String, param: String?) : super(orig.parser, orig.text!!, orig.modifier) {
        this.blockType = blockType
        this.param = param
        this.text = param
    }
}


interface BlockEventListener : AroundEventListener<BlockEvent>
